### DISCUSSION TITLE
Sharepoint automation help needed

### DISCUSSION THREAD

**Discussion by kgandrees** [community/community#52198] [Programming Help] [answered]
### Body

I'm new to working with Sharepoint and I've been tasked with working on a way to automate adding a Watermark to any document added to a file on a Sharepoint site.

Is this task possible?  If so, how would I do it?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by BREN0sx** [comment]
Yes! There are several ways to do this, however the best options will depend on your technical knowledge and your SharePoint environment.

For this, workflows in SharePoint Designer can be used to add the watermark. Another way is to use SharePoint Online Management Shell and PowerShell to automate the task.

You can also use third-party services that can be integrated with SharePoint. Perhaps a more practical way

Hope this helps!! 😉

**Reply by kgandrees** [reply]
Thanks.  I'm glad it can be done.  We are trying to avoid third-party additions.  Could you give me a rough idea of the process that would be used in Sharepoint Designer?  I'm not a developer and I'm new to Sharepoint, so any help you can provide is appreciated.

**Reply by BREN0sx** [reply, marked as answer]
In SharePoint Designer you must connect to your SharePoint site. You can create a workflow for the document library you want to add the watermark. Remember to make sure the workflow runs whenever an item is added or updated in the document library.

To allow you to modify the file that has been added to the documents, the "Edit Files" action must be added. Also, the "run a script" action to the previous action.

In the script you must add the necessary code to insert the watermark. It may vary depending on the type you want to use.
For example, if you want to use an image as a watermark, you can add code that overlays the document image.

It is important to note that the process may vary depending on the version of SharePoint you are using and your specific SharePoint site settings. Also, you will need programming skills to create the script that will add the watermark.

**Reply by kgandrees** [reply]
Thanks a ton.

**Reply by kgandrees** [reply]
I can't find an "Edit Files" action.  Is it subset in something else?

**Reply by kgandrees** [reply]
Is it not available on Sharepoint 2013?
